providers/class.two-factor-provider.php

```diff
@@ -71,13 +71,21 @@ public function print_label() {
 	 * @param string|array $chars Valid auth code characters.
 	 * @return string
 	 */
-	public function get_code( $length = 8, $chars = '1234567890' ) {
+	public function get_code( $length = 8, $chars = null ) {
+		// If no characters are passed in, just make it numeric.
+		if ( empty( $chars ) ) {
+			// A length of 3, will be between 0 and ( 10^3 - 1 ) or 999
+			$code = random_int( 0, ( pow( 10, $length ) - 1 ) );
+			return str_pad( $code, $length, '0', STR_PAD_LEFT );
+		}
 		$code = '';
 		if ( ! is_array( $chars ) ) {
 			$chars = str_split( $chars );
 		}
+		// Make sure it isn't an associative array which could mess up the key lookup.
+		$chars = array_values( $chars );
 		for ( $i = 0; $i < $length; $i++ ) {
-			$key = array_rand( $chars );
+			$key = random_int( 0, sizeof( $chars ) - 1 );
```

Comment:
I like the idea of better randomness, but `random_int()` isn't available [in PHP < 7.0](http://php.net/manual/en/function.random-int.php).

We could bundle [random_compat](https://github.com/paragonie/random_compat) with the project to maintain support for older versions of PHP. Otherwise, this change will bump the minimum version requirement.

Reply:
I mean, Core does bundle a polyfill for `random_int()` -- https://github.com/WordPress/WordPress/blob/0a81bcf4fffef460f22fda3fe5ccdf0376d1e3d9/wp-includes/random_compat/random_int.php

Reply:
As, same lib already there :-) Carry on.